src/service/stateTuple/logic/ArcballCameraControllerLogicService.re

```diff
@@ -185,4 +185,60 @@ let unbindGameViewActiveCameraArcballCameraControllerEvent = engineState =>
        engineState =>
          _checkSceneAllArcballCameraControllersNotBindEvent(engineState),
        StateEditorService.getStateIsDebug(),
-     );
\ No newline at end of file
+     );
+
+let setEditorCameraFocusTargetGameObject =
+    (distance, isCurrentSceneTreeNodeCanBeOperate, editorState, engineState) => {
+  WonderLog.Contract.requireCheck(
+    () =>
+      WonderLog.(
+        Contract.(
+          Operators.(
+            test(
+              Log.buildAssertMessage(
+                ~expect=
+                  {j|the editor camera should has arcballCameraController component|j},
+                ~actual={j|not|j},
+              ),
+              () =>
+              editorState
+              |> SceneViewEditorService.unsafeGetEditCamera
+              |. GameObjectComponentEngineService.hasArcballCameraControllerComponent(
+                   engineState,
+                 )
+              |> assertTrue
+            )
+          )
+        )
+      ),
+    StateEditorService.getStateIsDebug(),
+  );
+
+  let editorCameraArcballControllerComponent =
+    editorState
+    |> SceneViewEditorService.unsafeGetEditCamera
+    |. GameObjectComponentEngineService.unsafeGetArcballCameraControllerComponent(
+         engineState,
+       );
+
+  isCurrentSceneTreeNodeCanBeOperate ?
+    engineState
+    |> ArcballCameraEngineService.setArcballCameraControllerTarget(
+         editorCameraArcballControllerComponent,
+         editorState
+         |> SceneTreeEditorService.unsafeGetCurrentSceneTreeNode
+         |. GameObjectComponentEngineService.unsafeGetTransformComponent(
+              engineState,
+            )
+         |. TransformEngineService.getPosition(engineState),
+       )
+    |> ArcballCameraEngineService.setArcballCameraControllerDistance(
+         distance,
+         editorCameraArcballControllerComponent,
+       ) :
+    engineState
+    |> ArcballCameraEngineService.setArcballCameraControllerTarget(
```

Comment:
TODO feat: should set distance

Reply:
if currentGameObject is scene or don't have it, we don't need set distance